betty/cache.py

```diff
@@ -1,35 +1,456 @@
+
+    async def _set(self, cache_item_id: str, value: CacheItemValueT, *, modified: int | float | None = None) -> None:
+        raise NotImplementedError
+
+    @asynccontextmanager
+    async def getset(self, cache_item_id: str) -> AsyncIterator[tuple[CacheItemT | None, _CacheItemValueSetter[CacheItemValueT]]]:
+        async def _setter(value: CacheItemValueT) -> None:
+            await self._set(cache_item_id, value)
+        async with await self._lock(cache_item_id):
+            yield await self._get(cache_item_id), _setter
+
+    async def delete(self, cache_item_id: str) -> None:
+        async with await self._lock(cache_item_id):
+            await self._delete(cache_item_id)
+
+    async def _delete(self, cache_item_id: str) -> None:
+        raise NotImplementedError
 
     async def clear(self) -> None:
+        async with MultiLock(*self._locks.values()):
+            await self._clear()
+        logger = logging.getLogger(__name__)
+        if self.scopes:
+            logger.info(self._localizer._('"{scope}" cache cleared.').format(scope='.'.join(self.scopes)))
+        else:
+            logger.info(self._localizer._('All caches cleared.'))
+
+    async def _clear(self) -> None:
+        raise NotImplementedError
+
+
+class FileCacheItem(CacheItem[CacheItemValueT], Generic[CacheItemValueT]):
+    def __init__(
+        self,
+        id: str,
+        cache: _FileCache[Any],
+        modified: int | float,
+    ):
+        self._id = id
+        self._cache = cache
+        self._modified = modified
+        self._value: CacheItemValueT
+        self._value_loaded = False
+
+    @property
+    def id(self) -> str:
+        return self._id
+
+    @property
+    def scopes(self) -> Sequence[str]:
+        return self._cache.scopes
+
+    async def value(self) -> CacheItemValueT:
+        if not self._value_loaded:
+            async with aiofiles.open(self._cache._cache_item_file_path(self.id), 'rb') as f:
+                value_bytes = await f.read()
+            self._value = self._load_value(value_bytes)
+            self._value_loaded = True
+        return self._value
+
+    def _load_value(self, value_bytes: bytes) -> CacheItemValueT:
+        raise NotImplementedError
+
+    @property
+    def modified(self) -> int | float:
+        return self._modified
+
+
+class PickledFileCacheItem(FileCacheItem[CacheItemValueT], Generic[CacheItemValueT]):
+    def _load_value(self, value_bytes: bytes) -> CacheItemValueT:
+        return loads(value_bytes)  # type: ignore[no-any-return]
+
+
+class BinaryFileCacheItem(FileCacheItem[bytes]):
+    def _load_value(self, value_bytes: bytes) -> bytes:
+        return value_bytes
+
+    @property
+    def path(self) -> Path:
+        return self._cache._cache_item_file_path(self.id)
+
+
+@deprecated(f"This class is deprecated, and exists only for App.cache's backwards compatibility. Use {Cache} instead.")
+class FileCache(Cache[CacheItem[Any], Any]):
```

Comment:
This must not extend any classes, and instead declare its own `clear()` method. Anything else is a BC break.